src/validation.cpp

```diff
@@ -5863,3 +5867,29 @@ Chainstate& ChainstateManager::GetChainstateForIndexing()
     // indexed.
     return (this->GetAll().size() > 1) ? *m_ibd_chainstate : *m_active_chainstate;
 }
+
+std::pair<int, int> ChainstateManager::GetPruneRange(const Chainstate& chainstate, int last_height_can_prune)
+{
+    if (chainstate.m_chain.Height() <= 0) {
+        return {0, 0};
+    }
+    int prune_start{0};
+
+    if (this->GetAll().size() > 1 && m_snapshot_chainstate.get() == &chainstate) {
+        // Leave the blocks in the background IBD chain alone if we're pruning
+        // the snapshot chain.
+        prune_start = *Assert(GetSnapshotBaseHeight()) + 1;
+    }
+
+    int max_prune = chainstate.m_chain.Height() - static_cast<int>(MIN_BLOCKS_TO_KEEP);
```

Comment:
51622e9c1f7ed16f65844817a4a5e9bcbc157b4f

it's better to have std::max at this line, not below. This way I spend much less time thinking about the underflow, so does a future reviewer.